prov/sockets/src/sock_poll.c

```diff
@@ -151,6 +182,18 @@ static int sock_poll_close(fid_t fid)
 		p = head->next;
 		list_item = container_of(p, struct sock_fid_list, entry);
 		dlist_remove(p);
+		switch (list_item->fid->fclass) {
+		case FI_CLASS_CQ:
+			cq = container_of(list_item->fid, struct sock_cq, cq_fid);
+			atomic_dec(&cq->ref);
+			break;
+		case FI_CLASS_CNTR:
+			cntr = container_of(list_item->fid, struct sock_cntr, cntr_fid);
+			atomic_dec(&cntr->ref);
+			break;
+		default:
+			break;
+		}
```

Comment:
This code segment is shared with sock_poll_del -- please merge to common code, or call sock_poll_del directly from sock_poll_close.